From 6eba505f4c0f2db14abdb922fca568e8729e9661 Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Thu, 12 Jul 2018 17:21:57 +0700
Subject: [PATCH 040/475] Revert "mmap: relax file size limit for regular
 files"

This reverts commit 6ea1dc96a03a ("mmap: relax file size limit
	for regular files")

Signed-off-by: Thao M. Hoang <thaohm2@fsoft.com.vn>
Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 mm/mmap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/mmap.c b/mm/mmap.c
index ddc63e1..fe948d8 100644
--- a/mm/mmap.c
+++ b/mm/mmap.c
@@ -1291,7 +1291,7 @@ static inline int mlock_future_check(struct mm_struct *mm,
 static inline u64 file_mmap_size_max(struct file *file, struct inode *inode)
 {
 	if (S_ISREG(inode->i_mode))
-		return MAX_LFS_FILESIZE;
+		return inode->i_sb->s_maxbytes;
 
 	if (S_ISBLK(inode->i_mode))
 		return MAX_LFS_FILESIZE;
-- 
2.7.4

